### PR TITLE
Fix component list overflow

### PR DESCRIPTION
**Pull request by nummi**
This PR addresses two issues:

1. Clipping of a component background color that extends past the list bounds
    - This is addressed by adding a child element that has the same background color as the item
1. Component actions being hidden when the component extends past the list bounds
    - This is addressed by using `position: sticky` on the actions to pin them to the right side.

---

ℹ️ All additions to `app/styles/utils.scss ` are based off of Tailwind.

### CODE REVIEW

File: app/styles/base.scss [hunk 1]
```diff
@@ -14,6 +14,7 @@
   --unit10: calc(var(--unit1) * 10);
   --unit11: calc(var(--unit1) * 11);
   --unit12: calc(var(--unit1) * 12);
+  --transparent: transparent;
```
Comment (nummi) [inline app/styles/base.scss]: Linter was screaming at me :-|

File: app/styles/utils.scss [hunk 2]
```diff
@@ -65,7 +84,12 @@ $units: 0, 1, 2, 3, 4;
 .text-left { text-align: left; }
 .text-center { text-align: center; }
 .text-right { text-align: right; }
+.whitespace-no-wrap { white-space: nowrap; }
 
+.bg-transparent { background-color: var(--transparent); }
```
Comment (nummi) [inline app/styles/utils.scss]: Linter was screaming at me :-|

File: app/styles/component_tree.scss [hunk 3]
```diff
@@ -159,10 +133,6 @@
       fill: var(--focus-text);
     }
   }
-
-  .component-tree-item__action.disabled {
-    opacity: 0.65;
-  }
```
Comment (nummi) [inline app/styles/component_tree.scss]: This functionality went away at some point. 👋

File: app/templates/components/component-tree-item.hbs [hunk 4]
```diff
@@ -1,106 +1,109 @@
 <div
   style={{@item.style}}
   class="
-    component-tree-item
+    component-tree-item relative flex items-center mx-1 rounded
+    {{if @item.hasInstance "cursor-pointer" "cursor-default"}}
     {{if @item.isComponent "component-tree-item--component"}}
-    {{if @item.hasInstance "component-tree-item--has-instance"}}
-    {{if @item.isSelected "component-tree-item--selected"}}
-    {{if @item.isHighlighted "component-tree-item--highlighted"}}"
```
Comment (nummi) [inline app/templates/components/component-tree-item.hbs]: This was originally used to subtly highlight children of a selected component. At some point that changed and was being used for the hover state. There was a visible lag.

File: app/styles/component_tree.scss [hunk 5]
```diff
@@ -123,17 +101,7 @@
 
 /**
  * Modifier
- * highlighted - children of selected component
- */
-
-.component-tree-item.component-tree-item--highlighted {
-  background-color: var(--component-highlighted-bg);
-  border-radius: 0;
-}
```
Comment (nummi) [inline app/styles/component_tree.scss]: This was originally used to subtly highlight children of a selected component. At some point that changed and was being used for the hover state. There was a visible lag.

File: app/templates/components/component-tree-item.hbs (hunk 4 above)
Comment (rwwagner90) [reply app/templates/components/component-tree-item.hbs]: Do we still highlight the children?

Comment (nummi) [reply app/templates/components/component-tree-item.hbs]: Nope. Not sure if we should consider this a regression or pretend it never existed :-P

Comment (rwwagner90) [reply app/templates/components/component-tree-item.hbs]: When was the last time it worked? If it worked right before the Octane stuff, we should probably fix it.

Comment (nummi) [reply app/templates/components/component-tree-item.hbs]: I think it got lost in the flurry of Octane changes.

**Comment by nummi** [comment]
Fixed odd alignment.


screenshot

**Comment by nummi** [comment]
Adding a gradient so the clipping doesn't look odd:

image

**Comment by rwwagner90** [comment]
@chancancode looks like we have some failures. Are all the ember-try runs expected to run and pass? It does not seem to be a required check anymore, and it probably should be.

**Comment by chancancode** [comment]
Failure is due to #1130. It is required for merging things into _master_, but this PR is not targeting master so the branch protection rules did not apply.

**Pull request merged**